lumicks/pylake/kymotracker/detail/gaussian_mle.py

```diff
@@ -141,30 +146,38 @@ def gaussian_mle_1d(
     if fixed_background is not None and fixed_background <= 0:
         raise ValueError("Fixed background should be larger than zero.")
 
+    initial_position = x[np.argmax(photon_count)] if initial_position is None else initial_position
+    initial_position = np.atleast_1d(initial_position)
+
     expectation_fun = partial(
         peak_expectation_1d,
         x,
-        **({"background": fixed_background} if fixed_background is not None else {}),
+        fixed_background=fixed_background,
         pixel_size=pixel_size,
+        n_components=initial_position.size,
     )
     derivatives_fun = partial(
-        peak_expectation_1d_derivatives
-        if fixed_background is None
-        else peak_expectation_1d_derivatives_fixed_background,
+        peak_expectation_1d_derivatives,
         x,
+        fixed_background=fixed_background,
         pixel_size=pixel_size,
+        n_components=initial_position.size,
     )
-    initial_guess = (
-        np.max(photon_count) / pixel_size * np.sqrt(2 * np.pi * initial_sigma ** 2),
-        initial_position if initial_position is not None else x[np.argmax(photon_count)],
-        initial_sigma,
-        *([] if fixed_background is not None else [1]),
-    )
-    bounds = (
-        (0.01, None),
-        (np.min(x), np.max(x)),
-        (pixel_size, 10 * pixel_size),
-        *([] if fixed_background is not None else [(np.finfo(float).eps, None)]),
-    )
+
+    amp_estimate = np.max(photon_count) / pixel_size * np.sqrt(2 * np.pi * initial_sigma ** 2)
+    initial_guess, bounds = [], []
+    for init_pos in initial_position:
+        initial_guess.extend([amp_estimate / initial_position.size, init_pos, initial_sigma])
+        bounds.extend([(0.01, None), (np.min(x), np.max(x)), (pixel_size, 10 * pixel_size)])
+    if fixed_background is None:
+        initial_guess.append(1)
+        bounds.append((np.finfo(float).eps, None))
+
     result = _mle_optimize(initial_guess, expectation_fun, derivatives_fun, photon_count, bounds)
+
+    # Pack the results
+    background = result.x[-1] if fixed_background is None else fixed_background
+    result = tuple(
+        (*param, background) for param in result.x[: initial_position.size * 3].reshape(-1, 3)
+    )
     return result
```

Comment:
I think it makes sense to just return the tuple directly, it's clear from the comment what's going on

Reply:
Ah yes. Done.